aten/src/ATen/native/Distance.cpp

```diff
@@ -24,6 +24,20 @@ Tensor pdist(const Tensor& self, const double p) {
   return at::_pdist_forward(self.contiguous(), p);
 }
 
+Tensor euclidean_dist_out(const Tensor& x1, const Tensor& x2) {
+  Tensor x1_norm = x1.pow(2).sum(-1, true);
+  Tensor x2_norm = x2.pow(2).sum(-1, true);
+  Tensor result;
+  if (x1.dim() == 2) {
+    result = at::addmm(x2_norm.transpose(-2, -1), x1, x2.transpose(-2, -1), 1, -2);
+  } else {
+    result = at::baddbmm(x2_norm.transpose(-2, -1), x1, x2.transpose(-2, -1), 1, -2);
+  }
+  result.add_(x1_norm);
```

Comment:
Implementations in #15253 had a call filtering negative values, and it is necessary, it is possible to get negative distances with this approach because of floating point accuracy.

Reply:
yep, I changed this